prometheus/go_collector_latest_test.go

```diff
@@ -227,9 +251,15 @@ func TestMemStatsEquivalence(t *testing.T) {
 	for i := 0; i < msRealValue.NumField(); i++ {
 		fr := msRealValue.Field(i)
 		ff := msFakeValue.Field(i)
-		switch typ.Kind() {
```

Comment:
Fun fact. This test was noop, because `typ.Kind` was always `struct` - we want field kind instead 🙈

Reply:
oops! good catch.